binutils.mk

```diff
@@ -0,0 +1,113 @@
+binutils:
+	@echo "Using previously built binutils."
+else
+binutils: binutils-setup $(patsubst %,%-binutils-target,$(BINUTILS_TARGETS))
+	touch $(BUILD_WORK)/binutils/.build_complete
+endif
+
+%-binutils-target: binutils-setup gettext
+	target=$$(echo $@ | rev | cut -d- -f3- | rev); \
+	if [ "$$GNU_HOST_TRIPLE" = "$$target" ]; then \
+		binutils_extra_flags=--program-prefix=g; \
+	fi; \
+	if [ -f $(BUILD_WORK)/binutils/$$target/.build_complete ]; then \
+		echo "Using previously built $$target binutils"; \
+	else \
+		mkdir -p $(BUILD_WORK)/binutils/$$target; \
+		cd $(BUILD_WORK)/binutils/$$target && ../configure -C \
+			--build=$$($(BUILD_MISC)/config.guess) \
+			--host=$(GNU_HOST_TRIPLE) \
+			--prefix=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX) \
+			--libdir=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/$(GNU_HOST_TRIPLE)/$$target/lib \
+			--includedir=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/$(GNU_HOST_TRIPLE)/$$target/include \
+			--target=$$target \
+			$(BINUTILS_CONFARGS) $$binutils_extra_flags; \
+		$(MAKE) -C $(BUILD_WORK)/binutils/$$target; \
+		$(MAKE) -C $(BUILD_WORK)/binutils/$$target install \
+			DESTDIR=$(BUILD_STAGE)/binutils/$$target; \
+		touch $(BUILD_WORK)/binutils/$$target/.build_complete; \
+	fi
+
+binutils-package: binutils-stage
+	# binutils.mk Package Structure
+	for target in $(BINUTILS_TARGETS); do \
+		rm -rf $(BUILD_DIST)/binutils-$$target; \
+	done
+	rm -rf $(BUILD_DIST)/binutils{,-common}
+	mkdir -p $(BUILD_DIST)/binutils
+	mkdir -p $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)
+	
+	#  binutils.mk Prep binutils-*
+	for target in $(BINUTILS_TARGETS); do \
+		cp -r $(BUILD_STAGE)/binutils/$$target $(BUILD_DIST)/binutils-$$target; \
+	done
+	for target in $(BINUTILS_TARGETS); do \
+		for pkg in $(BUILD_DIST)/binutils-$$target; do \
+			for man in $$pkg/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/man/man1/*; do \
+				$(LN) -sf binutils-$$(basename $$man .1 | rev | cut -d- -f1 | rev ).1.zst $$man; \
+			done; \
+			rm -rf $$pkg/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/{locale,info}; \
+		done; \
+	done
+	cp -r $(BUILD_STAGE)/binutils/x86_64-linux-gnu/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)
+	for man in $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/man/man1/*; do \
+	mv $$man $(BUILD_DIST)/binutils-common/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/share/man/man1/binutils-$$(basename $$man .1 | rev | cut -d- -f1 | rev ).1; \
+	done
+	# binutils.mk Prep binutils
+	# No architecure-specific control mechanism in PACK, sorry.
+	$(SED) -e "s/@TARGET@/$$($(SED) -e 's/_/-/g' <<< "$(GNU_HOST_TRIPLE)")/g" $(BUILD_INFO)/binutils-native.control.in > $(BUILD_INFO)/binutils.control
+	
+	# binutils.mk Sign
+	for target in $(BINUTILS_TARGETS); do \
+		$(call SIGN,binutils-$$target,general.xml);\
+	done
+	
+	# Make control files with:
+	# sed "s/@TARGET@/$(sed 's/_/-/g' <<< "$target")/g" binutils.control.in > binutils-$target.control
```

Comment:
I know I added this line in just as a comment so adding control files is easy, but you should change it to generate the control files on the fly, simple to how you generate the native one.